DiamondLightSource/python-artemis
# 315 split fgs communicator

Fixes #315 

Separates the FGSCommunicator into three callback classes, adds a collection for easy subscription and handling the connection between the ispyb and zocalo components

### To test:
1. Check the changes make sense
2. Check the new test makes sense
3. Confirm tests pass
